internal/parser/complex_value.go

```diff
@@ -0,0 +1,135 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package parser
+
+import (
+	"fmt"
+	"math"
+	"strconv"
+	"strings"
+	"unicode"
+
+	"github.com/datacommonsorg/mixer/internal/base"
+)
+
+// ParseComplexValue parses a complex value string into node.
+//
+// [!!]This has minimal validation check, assuming the input has been checked
+// by the import tool.
+//
+// 1. a Quantity/QuantityRange value, coded as one of:
+//  [<unit> <val>]
+//  [<unit> <startval> <endval>]
+//  [<unit> - <endval>]
+//  [<unit> <startval> -]
+//
+// 2. a GeoCoordinates type, coded as one of:
+//  [LatLong <lat_value> <long_value>]
+//  [<lat_value> <long_value> LatLong]
+func ParseComplexValue(val string) string {
+	trimmed := strings.Trim(val, "[]")
```

Comment:
Does caller trim space?

Reply:
Yes, caller (which will be in the csv loader) will trim for each cell)